recipes/_start.rb

```diff
@@ -5,11 +5,30 @@
 #
 # Ensures the splunk instance is running, and performs post-start tasks.
 
+ulimit_command = "ulimit -n #{node['splunk']['limits']['open_files']}"
+init_file_path = '/etc/init.d/splunk'
+restart_flag = !(File.exist?(init_file_path) && File.readlines(init_file_path).grep(/#{ulimit_command}/).any?)
+
 # We want to always ensure that the boot-start script is in place on non-windows platforms
 execute "#{node['splunk']['cmd']} enable boot-start -user #{node['splunk']['user']}" do
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new(init_file_path)
+    file.insert_line_after_match(/^RETVAL=\d$/, ulimit_command)
+    file.write_file
+  end
+  not_if { platform_family?('windows') }
+end
+
+ruby_block 'restart-splunk' do
```

Comment:
I am not sure I see the issue with the guard. Since the variable we are using in the guard is just plain ruby code, it should be evaluated at compile time so we should be checking if a change actually occurred between the past state and what will be applied in the insertion. 

As for renaming it I am fine with that, but it is touching the restart marker so restart-splunk seems to make more sense than check-for-ulimit-change since that is really just what the guard is doing. I am open to suggestions for the name of the resource.